src/core/execute.c

```diff
@@ -6870,6 +6870,17 @@ bool exec_context_has_encrypted_credentials(ExecContext *c) {
         return false;
 }
 
+int exec_context_add_default_dependencies(Unit *u, const ExecContext *c) {
+        assert(u);
+        assert(u->default_dependencies);
+
+        if (c && c->tty_vhangup)
```

Comment:
hmm, i'd probably go further here, and instead of linking this to tty_vhangup, maybe just order this so that if a service touches a tty at all we add the dep.

i.e. basically, export exec_context_needs_term() and rename it and use it here.

and then maybe add another tweak here: if we can determine that the tty definitely isn't the vconsole even omit the dep, but this would be the cherry on top.